app/services/hyku_addons/qa_select_service.rb

```diff
@@ -25,29 +26,31 @@ def initialize(authority_name, model: nil)
     private
 
       def tenant_authority_name
-        [@authority_name, tenant_locale].compact.join("-")
+        [@authority_name, tenant_locale].reject(&:blank?).join("-")
       end
 
       def model_authority_name
-        [@authority_name, model_name].compact.join("-")
+        [@authority_name, model_name].reject(&:blank?).join("-")
       end
 
       def model_tenant_authority_name
-        [@authority_name, model_name, tenant_locale].compact.join("-")
+        [@authority_name, model_name, tenant_locale].reject(&:blank?).join("-")
```

Comment:
We're repeating `.reject(&:blank?).join("-")` 3 times in this class, so that would indicate that we can move that to a method. `clean` is just a suggested name. 

```ruby 
      def model_tenant_authority_name
        clean([@authority_name, model_name, tenant_locale])
      end

      def clean(array)
          array.reject(&:blank?).join("-")
      end
```

Reply:
`prepare_to_string`, or something that describes what it does without being super verbose.